### PR TITLE
new nft implementation

### PR DESCRIPTION
**Pull request by LouiseMedova**
- removed `nft-example`, `non-fungible-token` and `nft` from `nft-marketplace` project
- library for non-fungible-token
- changed the example of `non-fungible-token` and slightly changed`nft-marketplace` according to changes in `non-fungible-token`

### CODE REVIEW

File: nft/.binpath [hunk 1]
```diff
@@ -0,0 +1 @@
+target/wasm32-unknown-unknown/release/nft_example.opt.wasm
```
Comment (shamilsan) [inline nft/.binpath:1]: No need to keep this auto-generated file in git.

File: nft/Cargo.toml [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
+codec = { package = "parity-scale-codec", version = "3.1.2", default-features = false, features = ["derive"] }
 scale-info = { version = "2.0.1", default-features = false, features = ["derive"] }
+hex = { version = "0.4.3", default-features = false, features = ["alloc"] }
+gear-contract-libraries = { path = "../gear-contract-libraries" }
+nft-io = { path = "io" }
+derive_traits = { path = "../gear-contract-libraries/utils/derive_traits" }
 
 [dev-dependencies]
-gtest = { git = "https://github.com/gear-tech/gear.git" }
+gtest = { git = "https://github.com/gear-tech/gear.git"}
```
Comment (shamilsan) [inline nft/Cargo.toml:18]:
```suggestion
gtest = { git = "https://github.com/gear-tech/gear.git" }
```

File: nft/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    }
+}
+
+gstd::metadata! {
+    title: "NFT",
+    init:
+        input : InitNFT,
+    handle:
+        input : NFTAction,
+        output : Vec<u8>,
```
Comment (shamilsan) [inline nft/src/lib.rs:72]: We tend to keep struct style.
```suggestion
        input: InitNFT,
    handle:
        input: NFTAction,
        output: Vec<u8>,
```

**Review by shamilsan — COMMENTED** [review]
Some issues have been already fixed in https://github.com/gear-tech/apps/pull/36

File: nft/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    }
+}
+
+gstd::metadata! {
+    title: "NFT",
+    init:
+        input: InitNFT,
+    handle:
+        input: NFTAction,
+        output: Vec<u8>,
```
Comment (fluiderson) [inline nft/src/lib.rs:72]: Maybe it's better to create some `Event` enum in the `nft-io` crate instead of this?

**Review by fluiderson — COMMENTED** [review]
And please resolve the 3 previous unresolved suggestions by shamilsan:
- https://github.com/gear-tech/apps/pull/47#discussion_r867464840
- https://github.com/gear-tech/apps/pull/47#discussion_r867464916
- https://github.com/gear-tech/apps/pull/47#discussion_r867464981

File: gear-contract-libraries/src/non_fungible_token/royalties.rs [hunk 5]
```diff
@@ -1,30 +1,29 @@
...
+    accounts: Payout,
     percent: u16,
 }
 
-pub type Payout = BTreeMap<ActorId, u128>;
-
 impl Royalties {
     pub fn validate(&self) {
-        // percent must be less then 100% (100 * 100)
+        // percent must be less than or equal to100% (100 * 100)
```
Comment (fluiderson) [inline gear-contract-libraries/src/non_fungible_token/royalties.rs:14]:
```suggestion
        // percent must be less than or equal to 100% (100 * 100)
```

**Review by fluiderson — APPROVED** [review]

**Comment by shamilsan** [comment]
Closing in favor of: https://github.com/gear-academy/non-fungible-token/pull/4

TODO: Need to adapt Dutch auction though (https://github.com/gear-academy/dutch-auction).

**Pull request closed without merging**